### PR TITLE
Support v2.13.0, v2.12.1, v2.12.0, v2.11.1 on amd64

### LINKED ISSUE #5: trial to support v2.13.0
Hello,

Thank you for providing such a repository, definitively a must have for c++ developer wanting to play with IA!

I'm trying to have tensorflow C++ api build from source with latest v2.13.0 Tensorflow inspiring from your docker file and could successfully compile tensorflow but I'm facing some issues at runtime executing hello-tensorflow

To build v2.13.0 with gpu support I've made a docker container using the official build images `FROM tensorflow/build:2.13-python3.11`
Then inside the container run:
`git clone --branch v2.13.0 --depth=1 https://github.com/tensorflow/tensorflow.git`

Export all the env vars `PYTHON_BIN_PATH, PYTHON_LIB_PATH ... CC_OPT_FLAGS, TF_SET_ANDROID_WORKSPACE` as it is done in your docker file (line 43-52) and run the `./configure` and finally run 
`bazel build --jobs 8 --config=cuda --config=opt --config=monolithic --verbose_failures tensorflow:libtensorflow_cc.so tensorflow:install_headers;`
which allow to successfully build tensorflow

Then I compiled protobuf from source using `https://github.com/protocolbuffers/protobuf/archive/v3.21.9.zip` using same steps of your docker file (line 81-93)

I could then compile the hello-tensorflow with a 
`g++ hello-flow.o -lprotobuf -L/my/path/lib/tensorflow -ltensorflow_cc -ltensorflow_framework`

The issues I faced are:
1. I've got to link with `-ltensorflow_framework` whereas I used `--config=monolithic`
2. when running ./hello-tensorflow I got this error
```
[libprotobuf ERROR external/com_google_protobuf/src/google/protobuf/descriptor_database.cc:642] File already exists in database: google/protobuf/any.proto
[libprotobuf FATAL external/com_google_protobuf/src/google/protobuf/descriptor.cc:1986] CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
terminate called after throwing an instance of 'google::protobuf::FatalException'
  what():  CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
Aborted (core dumped)
```

I was wondering if you would have any suggestion to address those issues?
And if you would have any plan to support TensorFlow v2.13.0 I would be happy to help.

Regards.

### PR DESCRIPTION
**Pull request by lreiher**
- explicitly add Dockerfiles to build devel images for 2.13.0, since they are not part of the TF repository anymore
- devel image Dockerfiles are based on 2.12.1 with some fixes to actually build
- patch in main Dockerfile ensures that libtensorflow_cc 2.13.0 actually builds
- extra devel Dockerfile allows to build 2.13.0 against CUDA 12.1 instead of 11.8
- final images have to be upgraded to GCC 11 in order to build sample application
- modify setups for 2.12.0, 2.12.1 in a similar way
- 2.11.1 is also supported now, for both amd64 and arm64
- arm64 images are not yet supported/tested; builds have been failed so far, issues will be created

Closes #5. Related to https://github.com/ika-rwth-aachen/libtensorflow_cc/issues/7.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -242,6 +249,10 @@ make 5-print-versions
 
 | Version | Architecture | Step 1 (CPU) | Step 2 (CPU) | Step 4 (CPU) | Step 1 (GPU) | Step 2 (GPU) | Step 4 (GPU) | Notes |
 | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | --- |
+| 2.13.0 | amd64 | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |  |
+| 2.12.1 | amd64 | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | missing image `tensorflow/tensorflow:2.12.1` |
+| 2.12.0 | amd64 | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |  |
+| 2.11.1 | amd64 | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: | - | - | missing image `nvidia/cuda:11.2.1-base-ubuntu20.04` |
```
Comment (jpbusch) [inline README.md:255]: please add information about arm64 build

File: docker/.Dockerfiles/2.12.0/dockerfiles/devel-cpu.Dockerfile
Comment (jpbusch) [inline docker/.Dockerfiles/2.12.0/dockerfiles/devel-cpu.Dockerfile:1]: Should we also add and build the dockerfiles for arm64 (cpu only)?
- for 2.12.0, 2.12.1, 2.13.0

File: docker/Dockerfile [hunk 2]
```diff
@@ -39,8 +39,15 @@ RUN if [ "${TF_VERSION}" = "2.10.0" ] || [ "${TF_VERSION}" = "2.10.1" ]; then \
...
+# fix GPU-version build issue in 2.12.0, 2.12.1, 2.13.0
+# https://github.com/tensorflow/tensorflow/issues/60398#issuecomment-1537040602
+COPY patch/ patch/
+RUN if [ "${TF_VERSION}" = "2.12.0" ] || [ "${TF_VERSION}" = "2.12.1" ] || [ "${TF_VERSION}" = "2.13.0" ]; then \
+        git apply patch/${TF_VERSION}/workspace2.bzl.patch; \
+    fi
+
 # configure compilation
-ARG TF_CUDA_COMPUTE_CAPABILITIES=5.3,6.0,6.1,7.0,7.2,7.5,8.0,8.6
+ARG TF_CUDA_COMPUTE_CAPABILITIES=6.0,6.1,7.0,7.5,8.0,8.6,8.9,9.0
```
Comment (jpbusch) [inline docker/Dockerfile:50]: defaulting to amd64 CCs?

File: docker/.Dockerfiles/2.12.0/dockerfiles/devel-cpu.Dockerfile
Comment (lreiher) [reply docker/.Dockerfiles/2.12.0/dockerfiles/devel-cpu.Dockerfile:1]: yes

**Review by jpbusch — APPROVED** [review]

**Pull request merged**